### PR TITLE
[MIG] hr_contract_multi_job: Migration from v11 to v12

### PR DESCRIPTION
**Pull request by luistorresm**
Module migrated to v12

Extra was fixed [this](https://github.com/OCA/hr/pull/482/files#diff-6cfed3ccfd0eca62d466bfc475fd3f98R33). To avoid a traceback in the compute method.

The order is:
First, the compute and after the constraint, 

Then, in the case that the contract has two jobs like the main job, the job_id is empty and after the constraint shows the raise that you do not can have two main jobs.

### CODE REVIEW

**Comment by luistorresm** [comment]
Note: Apply the last commit on v11, that is a fix

**Comment by alexey-pelykh** [comment]
This still WIP?

**Comment by luistorresm** [comment]
Hello @alexey-pelykh 

I have detected an issue when you try to set 2 jobs like the main job. Nut it also is in v10.

Then I hope to fix on this version and make correctly the PR,

I will try to complete this tonight.

**Comment by luistorresm** [comment]
Hello @alexey-pelykh @pedrobaeza 

Could you review this PR, please?

Regards

File: hr_contract_multi_job/tests/__init__.py [hunk 1]
```diff
@@ -0,0 +1 @@
+from . import test_hr_contract_multi_jobs
```
Comment (alexey-pelykh) [inline hr_contract_multi_job/tests/__init__.py:1]: * License should be in every file

File: hr_contract_multi_job/views/hr_contract_view.xml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+        <!-- Inherited Contract view -->
```
Comment (alexey-pelykh) [inline hr_contract_multi_job/views/hr_contract_view.xml:3]: * License should be in every file

**Review by alexey-pelykh — CHANGES_REQUESTED** [review]

**Comment by alexey-pelykh** [comment]
Can't wait to see this finalized and merged!

File: hr_contract_multi_job/views/hr_contract_view.xml (hunk 2 above)
Comment (luistorresm) [reply hr_contract_multi_job/views/hr_contract_view.xml:3]: Hello @alexey-pelykh 

This is new for me, do you know in which Doc I could read about this?

Comment (alexey-pelykh) [reply hr_contract_multi_job/views/hr_contract_view.xml:3]: Hm, actually, I'm not so sure, yet other reviewers pointed that out. I believe that @pedrobaeza would be much more appropriate to jump in here

**Review by alexey-pelykh — APPROVED** [review]

File: hr_contract_multi_job/views/hr_contract_view.xml (hunk 2 above)
Comment (pedrobaeza) [reply hr_contract_multi_job/views/hr_contract_view.xml:3]: Well, in XML files there is not the habit of adding the license, but they can be considered code somehow. If you see the template at https://github.com/OCA/maintainer-tools/blob/master/template/module/views/model_name_view.xml#L3, it's included, but I'm not usually forcing this.

Comment (alexey-pelykh) [reply hr_contract_multi_job/views/hr_contract_view.xml:3]: Right, I'd 👍 to that as would suggest to making it a good tradition

**Comment by alexey-pelykh** [comment]
@luistorresm do you need any help in this?

File: hr_contract_multi_job/models/hr_contract.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    @api.constrains('contract_job_ids')
+    def _check_one_main_job(self):
+        # if the contract has no job assigned, a main job
+        # is not required. Otherwise, one main job assigned is
+        # required.
+        for contract in self.filtered('contract_job_ids'):
+            main_jobs = contract.contract_job_ids.filtered('is_main_job')
+            if len(main_jobs) != 1:
+                raise UserError(
+                    _("You must assign one and only one job position "
```
Comment (vpa-odoo) [inline hr_contract_multi_job/models/hr_contract.py:46]: User Error message should be clear.
For example :  You can assign one job position for the main job position 
OR
You can not assign more then one job position as a main job position

**Review by vpa-odoo — CHANGES_REQUESTED** [review]

File: hr_contract_multi_job/models/hr_contract.py (hunk 3 above)
Comment (alexey-pelykh) [reply hr_contract_multi_job/models/hr_contract.py:46]: @vpa-odoo IMO, it's not a question to migration. Yet can be solved as a separate commit in recent versions

File: hr_contract_multi_job/models/hr_contract_job.py [hunk 4]
```diff
@@ -15,15 +15,13 @@ class HrContractJob(models.Model):
...
-    name = fields.Char(string='Job', related='job_id.name', index=True)
+    name = fields.Char(string='Job Name', related='job_id.name', index=True)
     job_id = fields.Many2one(
         'hr.job',
-        string='Job',
         required=True,
         ondelete='cascade')
     contract_id = fields.Many2one(
         'hr.contract',
-        string='Contract',
```
Comment (alexey-pelykh) [inline hr_contract_multi_job/models/hr_contract_job.py]: @luistorresm any particular reason for removing the string?

File: hr_contract_multi_job/models/hr_contract_job.py [hunk 5]
```diff
@@ -15,15 +15,13 @@ class HrContractJob(models.Model):
     _name = 'hr.contract.job'
     _description = 'Relational object between contract and job'
 
-    name = fields.Char(string='Job', related='job_id.name', index=True)
+    name = fields.Char(string='Job Name', related='job_id.name', index=True)
     job_id = fields.Many2one(
         'hr.job',
-        string='Job',
```
Comment (alexey-pelykh) [inline hr_contract_multi_job/models/hr_contract_job.py]: @luistorresm any particular reason for removing the string?

**Comment by alexey-pelykh** [comment]
@luistorresm if possible, please also split README.rst to multiple files as per https://github.com/OCA/maintainer-tools/tree/master/template/module/readme

**Comment by alexey-pelykh** [comment]
@luistorresm are there plans in place for addressing the feedback?

File: hr_contract_multi_job/models/hr_contract_job.py (hunk 4 above)
Comment (luistorresm) [reply hr_contract_multi_job/models/hr_contract_job.py]: Hello @alexey-pelykh 

Sorry for the time.

In the v12 the string parameter is not necessary. Oddo takes the field name, and if is m2o, removes the `_id` or `_ids`

File: hr_contract_multi_job/models/hr_contract_job.py (hunk 5 above)
Comment (luistorresm) [reply hr_contract_multi_job/models/hr_contract_job.py]: The same here

**Comment by luistorresm** [comment]
> @luistorresm if possible, please also split README.rst to multiple files as per https://github.com/OCA/maintainer-tools/tree/master/template/module/readme

Done @alexey-pelykh 

Thanks

File: hr_contract_multi_job/__init__.py [hunk 6]
```diff
@@ -0,0 +1,3 @@
+# See LICENSE file for full copyright and licensing details.
```
Comment (alexey-pelykh) [inline hr_contract_multi_job/__init__.py]: `# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).`

Comment (luistorresm) [reply hr_contract_multi_job/__init__.py]: Done @alexey-pelykh

**Review by moylop260 — APPROVED** [review]
:+1:

**Comment by moylop260** [comment]
@alexey-pelykh 
Could you check last changes, please?

**Comment by alexey-pelykh** [comment]
@moylop260 I'd say code is good-enough, and I've approved the PR

**Review by max3903 — APPROVED** [review]
LGTM. No Test.

**Pull request merged**